### DISCUSSION TITLE
Doc: Folder structure for bigger applications

### DISCUSSION THREAD

**Discussion by akhilmhdh** [99designs/gqlgen#1894] [Ideas]
### What happened?

It would be really helpful if doc contains a section that briefs on handling bigger project's folder structure.
I feel this is the structure gqlgen is pointing to 

```
.
├── go.mod
├── go.sum
├── gqlgen.yml
├── graph
│   ├── generated
│   ├── schema
│        └── user.graphql
│        └── todo.graphql
│   ├── resolver
│        └── user.resolver.go
│        └── todo.resolver.go
│        └── resolver.go
│   └── model
│        ├── user.go
│        ├── todo.go
└── server.go
```

Can i organize something like the way in below that is split by business logic? If so kindly help me.  If you know a repo, i would glad to look into it.
Is the below against Golang way. (Someone who has heavy node experience 😄 )

```
.
├── go.mod
├── go.sum
├── gqlgen.yml
├── graph
│   ├── generated
│   ├── user
│        └── user.graphql
│        └── user.resolver.go
│        └── user.model.go
│   ├── todo
│        └── todo.graphql
│        └── todo.resolver.go
│        └── todo.model.go
└── resolver.go
└── schema.graphql
└── server.go
```

Some common issue's that i faced on organizing above structure is generated resolvers always goes to the folder dir mention in yml not with the schema folder.

### What did you expect?

A guide to organize the project folder structure

### versions
 - `gqlgen version`? v.14.0
 - `go version`? 1.17

**Comment by frederikhors** [comment]
I think this is more of a question than a bug or feature request. But we'll wait for someone more competent than me to decide.

**Comment by heisGarvit** [comment]
yes this is possbile

In the user.resolver.go add

`type Resolver struct{}`

and then in resolver.go


    type Resolver struct {
	    *user.Resolver
	    *todo.Resolver
    }
